### PR TITLE
Add optional arg to `serve` for extra paths to watch for changes

### PR DESCRIPTION
**Pull request by rparrett**
This PR implements a feature discussed in #2741.

Sanity check:

* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes

* [X] Are you doing the PR on the `next` branch?
* [X] Have you created/updated the relevant documentation page(s)?
  
  There does seem to be some existing documentation related to the args for `serve`, but it is not comprehensive, and I suspect this isn't important enough to include there.

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -89,6 +89,10 @@ pub enum Command {
         /// Default append port to the base url.
         #[clap(long)]
         no_port_append: bool,
+
+        /// Extra path to watch for changes, relative to the project root.
+        #[clap(long)]
+        extra_watch_path: Vec<String>,
```
Comment (Keats) [inline src/cli.rs:95]: Should it be `extra_watch_paths` since it's a vec?

File: src/fs_utils.rs [hunk 2]
```diff
@@ -17,6 +17,7 @@ pub enum ChangeKind {
     StaticFiles,
     Sass,
     Config,
+    Unknown,
```
Comment (Keats) [inline src/fs_utils.rs]: let's call it AdditionalDirectory or something similar, it's not strictly unknown

File: src/cmd/serve.rs [hunk 3]
```diff
@@ -810,6 +816,12 @@ pub fn serve(
                                 site = s;
                             }
                         }
+                        ChangeKind::Unknown => {
+                            // Probably a user-provided watch location. We can't know exactly what to update.
```
Comment (Keats) [inline src/cmd/serve.rs]: can we add an info message like (text can change obviously):

```
console::info("-> {full_path} changed. Recreating whole site.");
```

**Review by Keats — CHANGES_REQUESTED** [review]
Just some nits

File: src/cli.rs (hunk 1 above)
Comment (rparrett) [reply src/cli.rs:95]: It's a vec, but the user would multiple paths by repeating the argument.
```bash
zola serve --extra_watch_path="a" --extra_watch_path="b"
```
So I think the singular makes the most sense. There's some precedent for this, e.g.
```bash
curl --header "Content-Type: application/json" --header "Authorization: Bearer token" https://api.example.com
grep --regexp="pattern1" --regexp="pattern2" file.txt
```
Happy to change it if you prefer though.

File: src/fs_utils.rs (hunk 2 above)
Comment (rparrett) [reply src/fs_utils.rs]: I went with `ExtraPath` to match other names used in this feature and added a doc comment to `ChangeKind`. Let me know if you'd prefer something else though.

File: src/cli.rs (hunk 1 above)
Comment (Keats) [reply src/cli.rs:95]: That's fine, I thought it was `--extra_watch_paths=something,something_else` with clap being smart enough to split on the `,`

**Review by Keats — APPROVED** [review]

**Comment by rparrett** [comment]
Thanks!

**Pull request merged**